### DISCUSSION TITLE
designing global search using mif2

### DISCUSSION THREAD

**Discussion by Fuhan-Yang** [kingaa/pomp#196] [Q&A] [answered]
I have a general question about global search using `mif2`. From the materials in the [SBIED material](https://kingaa.github.io/sbied/mif/Q_fit_seir.html) and other case studies, the global search is conducted several times with the increase of `Nmif` and the number of initial parameters. By increasing the number of initial parameters, we can identify the likelihood surface better and better. But could you comment if the benefits of increasing initial parameters are worthwhile when more `Nmif` (and maybe more `Np`) will be used? For example, say our goal is to get MLE. At run-level 3, we've known that the range of the parameters gets smaller after two runs, can we shrink the parameter range (and decrease the number of initial parameters) based on the estimates from the last runs, and increase `Nmif` and `Np` to have better performance for each `mif2` run? In this way, we may save some computational power.

**Comment by kingaa** [comment, marked as answer]
This is a good question about global search strategies.  There is no foolproof global search strategy, nor can there ever be one, as the so-called *No Free Lunch Theorems* establish.  The elements of effective strategies based on iterated filtering combine some number of starting guesses, together with some effort at searching locally from each guess.  There are inherent tradeoffs:

1. Increasing search effort generally leads to increased confidence that the maximum likelihood encountered is the global MLE.  The cost is computational.
2. Increasing the effectiveness of local search (for example, by increasing `Nmif` and `Np`, as well as random-walk intensities, makes each local search more effective, thus reducing the number of needed starting guesses.  But note that it is easier to parallelize across local searches than within them.

The precise location of the optimum strategy along the trade-off surface necessarily depends on the details of model and data.   With perfect knowledge of both, it would be feasible to find the optimum strategy.  It would also be unnecessary, since one would already know the MLE.

For these reasons, **pomp** provides you with tools to carry out your own customized search strategy.  In general, as your understanding of the system grows, you can save on computation, perhaps in the way you describe in your question.

One final remark: one's goal is never simply to find the MLE itself.  Such a point estimate is not very valuable without an estimate of its uncertainty.  For this reason, it is necessary to understand the shape of the likelihood surface, at least in a neighborhood of the MLE.  More generally, qualitative knowledge of the likelihood surface's shape is often more useful than quantitative numerical estimates alone.

**Reply by Fuhan-Yang** [reply]
Thanks for the insights, this is really helpful!
